go-controller/pkg/util/multi_network.go

```diff
@@ -682,17 +682,30 @@ func NewNetInfo(netconf *ovncnitypes.NetConf) (NetInfo, error) {
 	if netconf.Name == types.DefaultNetworkName {
 		return &DefaultNetInfo{}, nil
 	}
+	var ni NetInfo
+	var err error
 	switch netconf.Topology {
 	case types.Layer3Topology:
-		return newLayer3NetConfInfo(netconf)
+		ni, err = newLayer3NetConfInfo(netconf)
 	case types.Layer2Topology:
-		return newLayer2NetConfInfo(netconf)
+		ni, err = newLayer2NetConfInfo(netconf)
 	case types.LocalnetTopology:
-		return newLocalnetNetConfInfo(netconf)
+		ni, err = newLocalnetNetConfInfo(netconf)
 	default:
 		// other topology NAD can be supported later
 		return nil, fmt.Errorf("topology %s not supported", netconf.Topology)
 	}
+	if err != nil {
+		return nil, err
+	}
+	ipv4Mode, ipv6Mode := ni.IPMode()
+	if ipv4Mode && !config.IPv4Mode {
+		return nil, fmt.Errorf("network %s is attempting to use ipv4 subnets, which the cluster does not support", ni.GetNetworkName())
+	}
+	if ipv6Mode && !config.IPv6Mode {
+		return nil, fmt.Errorf("network %s is attempting to use ipv6 subnets, which the cluster does not support", ni.GetNetworkName())
```

Comment:
nit: how about the following rephrasing "network %q specifies IPv4 subnets but the cluster doesnt support IPv4"

Reply:
done